src/satosa/micro_services/ldap_attribute_store_no_pool.py

```diff
@@ -0,0 +1,195 @@
+"""
+SATOSA microservice that uses an identifier asserted by
+the home organization SAML IdP as a key to search an LDAP
+directory for a record and then consume attributes from
+the record and assert them to the receiving SP.
+"""
+
+from satosa.micro_services.base import ResponseMicroService
+from satosa.logging_util import satosa_logging
+from satosa.response import Redirect
+from satosa.exception import SATOSAError
+
+import copy
+import logging
+import ldap3
+import urllib
+
+from ldap3.core.exceptions import LDAPException
+
+from . ldap_attribute_store import *
+
+logger = logging.getLogger(__name__)
+
+class LdapAttributeStoreNoPool(LdapAttributeStore):
```

Comment:
Instead of inheriting and creating a new micro-service, can this be a config option in the existing code?

Reply:
I got issues to handle this kind of approach but we can try some better code to get this result. 
I don't know if these issues could depend by a different version of ldap3, I used ldap3==2.6.

Using the clientstrategy RESTARTABLE I didn't get a valid message_id from connection.search
````
message_id = connection.search(config['search_base'], search_filter, attributes=config['search_return_attributes'].keys())
````
We can do some better code generalization using a more detailed config file if you're interested to this, sure.

Reply:
Done in https://github.com/IdentityPython/SATOSA/pull/240/commits/121fc799fc76e89b8394b51992548e91bc3721df

Tested in my environment, actually we have two conflicts. The first one is trivial, the second one is more a programming style decision. You'll tell me what direction if you agree.

Beyond these the most important things are here https://github.com/IdentityPython/SATOSA/pull/240/files#diff-6885077c6ac78f71c00e9192de67a5ebR446 
and here https://github.com/IdentityPython/SATOSA/pull/240/files#diff-6885077c6ac78f71c00e9192de67a5ebR468

Regarding these lastest two, I don't know if this behaviour could be linked to ldap3 version used or from the client_strategy method (message_id). So, before merging I have to ask to you a simple test in an environment with `client_strategy=ldap3.REUSABLE`, as that code in the past was made to be working with it. After your feedback we could do the final refactor before merge if you agree this line.